### DISCUSSION TITLE
Importing bug - MSI line-of-business app

### DISCUSSION THREAD

**Discussion by Filisteos** [Micke-K/IntuneManagement#164] [General]
Hello @Micke-K, first of all, thank you for your work to make it easier for us as administrators to manage Intune.

I have found what is possibly a bug in the tool when importing MSI applications.
These are steps I have followed:

1. Export app from tenant 1
2. Add .MSI file to the folder where the .json is located
3. Import app to tenant 2

The import appears to be successful, but when I go to the assignments section I only have 2 options available:
- **Required** 
- _Available for enrolled devices._ This option is missing
- **Uninstall**

I go back to tenant 1 to check if it also happens here, but to my surprise the application has all 3 options, so there must be an error in the import.

I took a look at the .json and I see that the parameter "useDeviceContext" has the value "null", here is the first error, it should be "true" = "device context", in the intune panel this parameter having value " null" is displayed as if none of the "device context" and "user context" options were selected, this is a problem.

As a result of what happened I set the value to "true" in the .json, I imported the application again and now the "device context" parameter in Intune appears correctly selected, but the assignments tab continues to only show "Required" and "Uninstall ", the option "Available for enrolled devices" is still missing.

I have also verified that by creating the application manually, all 3 options do appear, so the problem has to be due to the export/import phase, 100%. Please, could you help me? I am at your disposal for any test. Thank you very much in advance.

**Comment by Micke-K** [comment]
Hello!

Is this important a LOB app or Win32 app (.intunewin)

Interesting issue. I've not seen this. I'll have a look at it and see if I can figure it out. 

Cheers!

**Reply by Filisteos** [reply]
At the moment it has only happened to me with LOB apps.
Another curious thing, if the exported application is in the user's context, the "useDeviceContext" parameter does not appear "null" and appears as it should, with the value "false" and also the 3 options are displayed correctly. The problem I mentioned in my previous comment only occurs when exporting an application whose installation is at the device level.
Thank you!

**Comment by Filisteos** [comment]
Hello @Micke-K, any news? If you need anything else let me know.  Thank you so much!

**Comment by Micke-K** [comment]
Hello @Filisteos,

I haven't gad a chance to look at this yet. It's been busy and there are a few things on the list to fix. I'll try to make some time for this soon. 

Cheers!

**Comment by Filisteos** [comment]
No worries. Thank you!

**Comment by Micke-K** [comment]
Hello!

This is a weird one. I spent a whole evening trying to understand it. I can replicate it but not fix it. 

I checked how it is done via the portal and tried the same json via code etc. but it still doesn't work.

I can see the Available assignment option until it commits the file upload. It is gone after that. 

I'll see if I can do some more troubleshooting another day.

Cheers!

**Reply by Filisteos** [reply]
Thanks for your time. I'm glad you managed to replicate the bug.

**Reply by Micke-K** [reply]
Hello!

Good news. I found the issue and I can now import system app and Available assignments are there. 

I need to do some more tests though because I've changed a few other things in the app upload. 

Cheers!

**Reply by Filisteos** [reply]
You are amazing! When you publish the new version I will try it!
I also have some suggestions that I think would greatly improve the tool, where can I send them? And one last question, any advice for learning to manage Azure from powershell like you do?

**Reply by Micke-K** [reply]
Thank you :) 

Please creat an issue for each feature request. I think you can tag them as well. I'll do that otherwise. I'm happy for any suggestions. As long as they are reasonable 😃. 

Regarding PiwerShell and Azure...it all depends on programming background and what you are trying to achieve. My suggestion is to start small and then add bit by bit. Like I did with my project. I started exporting Configuration Policies. Then I added more object to export. Then I looked at importing objects. Then I added compare etc. Add one function at the time. It takes time. 

I restarted the whole project and rewrote everything when I realised it was actually used. First version was messy and not very extendable. My coding style is a bit messy in general. I cone from a C/C++/C# background so my PowerShell code looks more like that. Sometimes I really regret using PowerShell but it is easy and more people know it. However, the way I implemented a fully modular solution in my project would probably confuse most people. It confuses even me sometimes :). 

Cheers!
